docs/src/main/asciidoc/neo4j-guide.adoc

```diff
@@ -0,0 +1,545 @@
+----
+package org.acme.neo4j;
+
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.neo4j.driver.Driver;
+
+@Path("fruits")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class FruitResource {
+
+    @Inject
+    Driver driver;
+}
+----
+
+=== Reading nodes
+
+Add the following method to the fruit resource:
+
+[source,java]
+.src/main/java/org/acme/neo4j/FruitResource.java
+----
+@GET
+public CompletionStage<Response> get() {
+    AsyncSession session = driver.asyncSession(); // <1>
+    return session
+        .runAsync("MATCH (f:Fruit) RETURN f ORDER BY f.name")  // <2>
+        .thenCompose(cursor ->  // <3>
+            cursor.listAsync(record -> Fruit.from(record.get("f").asNode()))
+        )
+        .thenCompose(fruits ->  // <4>
+            session.closeAsync().thenApply(signal -> fruits)
+        )
+        .thenApply(Response::ok) // <5>
+        .thenApply(ResponseBuilder::build);
+}
+----
+<1> Open a new, asynchronous session with Neo4j
+<2> Execute a query. This is a Cypher statement.
+<3> Retrieve a cursor, list the results and create `Fruits`.
+<4> Close the session after processing
+<5> Create a JAX-RS response
+
+Now start Quarkus in `dev` mode with:
+
+[source, shell]
+----
+mvn compile quarkus:dev
+----
+
+and retrieve the endpoints like this
+
+[source, shell]
+----
+curl localhost:8080/fruits
+----
+
+There are not any fruits, so create some:
+
+=== Creating nodes
+
+The `POST` method looks similar.
+It uses transaction functions of the driver:
+
+[source,java]
+.src/main/java/org/acme/neo4j/FruitResource.java
+----
+@POST
+public CompletionStage<Response> create(Fruit fruit) {
+    AsyncSession session = driver.asyncSession();
+    return session
+        .writeTransactionAsync(tx ->
+            tx.runAsync("CREATE (f:Fruit {name: $name}) RETURN f", Values.parameters("name", fruit.name))
+        )
+        .thenCompose(fn -> fn.singleAsync())
+        .thenApply(record -> Fruit.from(record.get("f").asNode()))
+        .thenCompose(persistedFruid -> session.closeAsync().thenApply(signal -> persistedFruid))
+        .thenApply(persistedFruid -> Response
+            .created(URI.create("/fruits/" + persistedFruid.id))
+            .build()
+        );
+}
+----
+
+As you can see, we are now using a Cypher statement with named parameters (The `$name` of the fruit).
+The node is returned, a fruit entity created and than mapped to a `201` created response.
+
+A curl request against this path may look like this:
+
+[source, shell]
+----
+curl -v -X "POST" "http://localhost:8080/fruits" \
+     -H 'Content-Type: application/json; charset=utf-8' \
+     -d $'{
+  "name": "Banana"
+}'
+----
+
+The response contains an URI that shall return single nodes.
+
+=== Read single nodes
+
+This time, we signal a read only transaction.
+This way, a Neo4j cluster could select a follower.
```

Comment:
`follower`?

Reply:
Good point. I would have to go into more detail here as it would fit into the guide.